internal/database/db.go

```diff
@@ -0,0 +1,94 @@
+package database
+
+import (
+	"fmt"
+	"github.com/cyneruxyz/address-book/gen/proto"
+	"github.com/cyneruxyz/address-book/internal/database/model"
+	"github.com/spf13/viper"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+var fieldModel = &model.AddressField{}
+
+type Database struct {
+	orm *gorm.DB
+}
+
+func New(conf *viper.Viper) (*Database, error) {
+	dsn := fmt.Sprintf(
+		"user=%s password=%s dbname=%s port=%s sslmode=%s TimeZone=%s",
+		conf.GetString("DB_USER"),
+		conf.GetString("DB_PASSWORD"),
+		conf.GetString("DB_NAME"),
+		conf.GetString("DB_PORT"),
+		conf.GetString("DB_SSLMODE"),
+		conf.GetString("DB_TIMEZONE"),
+	)
+
+	db, err := gorm.Open(postgres.New(
+		postgres.Config{
+			DSN:                  dsn,
+			PreferSimpleProtocol: true, // disables implicit prepared statement usage
+		}),
+		&gorm.Config{},
+	)
+
+	return &Database{db}, err
+}
+
+func (db *Database) Migrate() error {
+	return db.orm.AutoMigrate(fieldModel)
```

Comment:
Instead of using AutoMigrate it is a better practice to create migrations folder that is used when database is starting.
Example with golang-migrate CLI - https://medium.com/geekculture/db-migration-in-go-lang-d325effc55de
You can up migrations when your server is starting after connecting to the database